packages/web3torrent/src/clients/web3torrent-client.ts

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import WebTorrentPaidStreamingClient, {ExtendedTorrent} from '../library/web3torrent-lib';
+import {EmptyTorrent, Status, Torrent} from '../types';
+
+export const web3torrent = new WebTorrentPaidStreamingClient();
+export const WebTorrentContext = React.createContext(web3torrent);
+
+export const download: (torrent: any) => Promise<Torrent> = torrentData =>
+  new Promise(resolve =>
+    web3torrent.add(torrentData, torrent => resolve({...torrent, status: Status.Connecting}))
+  );
+
+export const upload: (files: any) => Promise<Torrent> = files =>
+  new Promise(resolve =>
+    web3torrent.seed(files as FileList, torrent => resolve({...torrent, status: Status.Seeding}))
+  );
```

Comment:
These functions should be tagged as `async`, since they're used with `await` further in the code.

Reply:
uh, yeah